tests/compiler/test_typing.py

```diff
@@ -380,3 +380,14 @@ def test_2243_override_optional(snippetcompiler) -> None:
 Child()
         """,
     )
+
+
+def test_print_number(snippetcompiler, capsys):
```

Comment:
Can you also add a test case for `int(1)` that it does not become `1.0`? Or does that already exist somewhere?

Reply:
I think this test is enough:
https://github.com/inmanta/inmanta-core/blob/fix-bug-number-cast/tests/compiler/test_typing.py#L131

Reply:
I think so. But perhaps we could extend it with an explicit `not any(isinstance(v, (bool, float)) for v in (x, y, z))`?